flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendTestBase.java

```diff
@@ -3902,6 +3906,84 @@ public void testParallelAsyncSnapshots() throws Exception {
 		}
 	}
 
+	@Test
+	public void testNonConcurrentSnapshotTransformerAccess() throws Exception {
+		Random rnd = new Random();
+		BlockerCheckpointStreamFactory streamFactory = new BlockerCheckpointStreamFactory(1024 * 1024);
+		AbstractKeyedStateBackend<Integer> backend = null;
+		try {
+			backend = createKeyedBackend(IntSerializer.INSTANCE);
+			InternalValueState<Integer, VoidNamespace, String> valueState = backend.createInternalState(
+				VoidNamespaceSerializer.INSTANCE,
+				new ValueStateDescriptor<>("test", StringSerializer.INSTANCE),
```

Comment:
Would be good to not only test with the `ValueStateDescriptor`. I think there might be a bug with the RocksDB list and map transformer.